### PR TITLE
Update component gallery with grouping and list view

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:

Hackathon project! I [added the component gallery](https://github.com/Khan/wonder-blocks/pull/2102) to the overview page
last year, and I've been wanting to make some updates to it for a while.

- Allow grouping by function rather than package
- Also added alphabetical grouping
- Added list view
- Added compact gallery view to it's easier to see more components
  at once
- Also added a seperate page for it in case people just never scrolled
  down to it in the overview page. 

Issue: none

## Test plan:
Storybook
- http://localhost:6061/?path=/docs/overview--docs
- http://localhost:6061/?path=/docs/gallery--docs

image

### CODE REVIEW

File: __docs__/components/gallery/component-gallery.tsx [hunk 1]
```diff
@@ -1,49 +1,125 @@
...
+                    <SingleSelect
+                        selectedValue={currentGroup}
+                        onChange={setCurrentGroup}
+                        // Placehoder is not used here
+                        placeholder=""
+                    >
+                        <OptionItem label="alphabet" value="alphabet" />
+                        <OptionItem label="function" value="function" />
+                        <OptionItem label="package" value="package" />
+                    </SingleSelect>
```
[the remark this answers is not in the document]
Comment (jandrade) [reply __docs__/components/gallery/component-gallery.tsx:51]: Also `LabelMedium tag="label"` would be useful :)

File: __docs__/components/gallery/component-gallery.tsx [hunk 2]
```diff
@@ -1,49 +1,125 @@
...
+
+                    {currentLayout === "grid" && (
+                        <>
+                            <Spring />
+                            <Switch
+                                checked={compactGridView}
+                                onChange={setCompactGridView}
+                            />
+                            <Strut size={spacing.xSmall_8} />
+                            <LabelMedium>Compact grid</LabelMedium>
```
Comment (jandrade) [inline __docs__/components/gallery/component-gallery.tsx:78]: **suggestion:** Same suggestion here. In fact, I wonder what would happen if you use `LabelMedium` as the parent container and include the label text and the `Switch` instance in it (maybe vertical centering fails?).

File: __docs__/components/gallery/groups.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import TooltipTile from "./tiles/tooltip-tile";
+
+export type GroupMap = {
+    name: string;
+    components: Array<
+        (props: {layout: "grid" | "list"; compactGrid: boolean}) => JSX.Element
+    >;
+};
+
+export const alphabetGroups: Array<GroupMap> = [
```
Comment (jandrade) [inline __docs__/components/gallery/groups.ts:39]: **praise:** This was a really nice improvement! thanks 🫶

**Review by jandrade — APPROVED** [review]
This is a really nice improvement! it really helps to visually understand better what components we have available in WB. Thanks for iterating on this and adding more options for displaying the info 👏

File: __docs__/components/gallery/component-gallery.tsx (hunk 2 above)
Comment (nishasy) [reply __docs__/components/gallery/component-gallery.tsx:78]: I tried this, but it looked like the Switch stopped working if I did that 🤔

File: __docs__/components/gallery/component-gallery.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nishasy) [reply __docs__/components/gallery/component-gallery.tsx:51]: Sorry for neglecting all these!

**Comment by beaesguerra** [comment]
@nishasy This is super cool! Thanks for adding this in! 🎉

**Pull request merged**